modules/custom/az_migration/src/Plugin/migrate/process/EntityEmbedProcess.php

```diff
@@ -0,0 +1,296 @@
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+    );
+
+    $instance->migrateLookup = $container->get('migrate.lookup');
+    $instance->migrateStub = $container->get('migrate.stub');
+    $instance->entityTypeManager = $container->get('entity_type.manager');
+    return $instance;
+  }
+
+  /**
+   * Generate a translated DOM element for the new embed.
+   *
+   * @param string $id
+   *   Source id (nid, etc.)
+   * @param string $type
+   *   Embed type, eg. node, file.
+   * @param string $tag
+   *   HTML Tag type to use for embed.
+   * @param string $view
+   *   Drupal view mode.
+   * @param \DOMDocument $dom
+   *   The active DOM.
+   * @param \DOMElement $element
+   *   The element to be replaced.
+   * @param string $migration
+   *   Migration name of id, for stubbing/lookup.
+   * @param string $storage
+   *   Storage controller for embedded item, eg. node or media.
+   *
+   * @return \DOMDocument
+   *   The new DOM element to use for replacement. NULL if none.
+   */
+  public function updateEmbedTag($id, $type, $tag, $view, \DOMDocument $dom, \DOMElement $element, $migration, $storage) {
+
+    if (empty($id) || empty($type) || empty($tag)) {
+      return NULL;
+    }
+
+    // Set up our replacement element.
+    $changed = $dom->createElement($tag);
+    $changed->setAttribute('data-entity-type', $type);
+    $ids = $this->migrateLookup->lookup($migration, [$id]);
+    if (empty($ids)) {
+      $ids = $this->migrateStub->createStub($migration, [$id]);
+    }
+    // We eventually found our id, by lookup or stubbing it.
+    if (!empty($ids)) {
+      $id = reset($ids);
+      if (!empty($id)) {
+        $eid = reset($id);
+        $entity = $this->entityTypeManager->getStorage($storage)->load($eid);
+        if ($entity) {
+          $changed->setAttribute('data-entity-uuid', $entity->uuid());
+        }
+      }
+    }
+
+    // Data alignment.
+    $align = $element->getAttribute('data-align');
+    if (!empty($align)) {
+      $changed->setAttribute('data-align', $align);
+    }
+
+    // Alt text.
+    $alt = $element->getAttribute('alt');
+    if (!empty($alt)) {
+      $changed->setAttribute('alt', $alt);
+    }
+
+    // Type specific attributes.
+    switch ($type) {
+      case 'file':
+        $changed->setAttribute('data-view-mode', $view);
+        break;
+
+      case 'node':
+        $changed->setAttribute('data-embed-button', "az_embed_content");
+        $changed->setAttribute('data-entity-embed-display', "view_mode:node.{$view}");
+        break;
+    }
+    return $changed;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+
+    $dom = @\DOMDocument::loadHTML($value, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD);
```

Comment:
Looks like calling `DOMDocument::loadHTML()` statically is deprecated and not supported in PHP 8.0.0+:
https://www.php.net/manual/en/domdocument.loadhtml.php#refsect1-domdocument.loadhtml-errors